### PR TITLE
Add support for template secrets

### PR DESCRIPTION
**Pull request by reddec**
Add support for template secrets.

Users can specify the structure and content of secrets based on their unique requirements using standard 
[Go templates](https://pkg.go.dev/text/template#hdr-Actions). This flexibility allows for a more tailored approach to
meeting the specific needs of different applications.

Available context:

| Variable    | Meaning                  |
|-------------|--------------------------|
| `.Host`     | Database host            |
| `.Role`     | Generated user/role name |
| `.Database` | Referenced database name |
| `.Password` | Generated role password  |

Example

```yaml
apiVersion: db.movetokube.com/v1alpha1
kind: PostgresUser
metadata:
  name: my-db-user
spec:
  role: username
  database: my-db
  secretName: my-secret
  privileges: OWNER
  template:
    PQ_URL: "host={{.Host}} user={{.Role}} password={{.Password}} dbname={{.Database}}"
```

Change is backward compatible: no need to update existing CRDs in cluster.

### CODE REVIEW

**Comment by reddec** [comment]
@hitman99 sorry for bothering you, but could you please review one more PR?

File: README.md [hunk 1]
```diff
@@ -173,6 +174,8 @@ spec:
   privileges: OWNER     # Can be OWNER/READ/WRITE
   annotations:          # Annotations to be propagated to the secrets metadata section (optional)
     foo: "bar"
+  template:             # Output secrets can be customized using standard Go templates
```
Comment (hitman99) [inline README.md:177]: I think that `template` is a bit misleading, would it be possible to change it to `secretTemplate` or something else that indicates that this fiel is used to template data in the secret

**Review by hitman99 — CHANGES_REQUESTED** [review]
I think it's a bit misleading to name that field `template`. From the name itself it's not clear that it is allowing to template the data in the secret

**Comment by reddec** [comment]
@hitman99 updated

**Review by hitman99 — APPROVED** [review]

**Pull request merged**